presto-main/src/main/java/com/facebook/presto/operator/TaskContext.java

```diff
@@ -103,20 +106,20 @@ public TaskContext(QueryContext queryContext,
         this.notificationExecutor = requireNonNull(notificationExecutor, "notificationExecutor is null");
         this.yieldExecutor = requireNonNull(yieldExecutor, "yieldExecutor is null");
         this.session = session;
-        taskStateMachine.addStateChangeListener(new StateChangeListener<TaskState>()
-        {
-            @Override
-            public void stateChanged(TaskState newState)
-            {
+        this.verboseStats = verboseStats;
+        this.cpuTimerEnabled = cpuTimerEnabled;
+    }
+
+    public void initialize()
```

Comment:
Add comment to explain why this is done this way (to avoid leaking this in constructor). Readers of this code won't be able to easily understand why this is not in constructor.

Reply:
There is a better way to do this.  You make the constructor and the initialize method private, and then add a factory method that call the constructor and the initialize.  This makes the state tracking simple.